providers/implementations/keymgmt/mac_legacy_kmgmt.c

```diff
@@ -89,6 +102,9 @@ int mac_key_up_ref(MAC_KEY *mackey)
 {
     int ref = 0;
 
+    if (!ossl_prov_is_running())
```

Comment:
??

Reply:
I thought there was a second, it might have been in the HW code paths.  I'll check if it can be reached unguarded -- if it can, it's equivalent to a `new` call and should be guarded.

Reply:
Both call paths here are from `signature/mac_legacy.c` and are already guarded so this isn't strictly required.  I'm in a quandary about leaving the guard in on the off chance something else calls it in the future and saving a couple of cycles by not.

The function `mac_key_up_ref()` isn't used in this file and is only used in `signature/mac_legacy.c` so it could move there and be made static.  However, it fits in really nicely with the `mac_key_new` and `mac_key_free` functions here.

I'll leave it in with a comment noting that it is for future safety rather than necessity.

Reply:
There are lots of guarded up_refs, they aren't all so closely guarded.  `kdf_data_up_ref()` is similar to this and I'm adding a similar comment.

The only unguarded up_ref calls are in `cipher_aes_siv_hw.c` and that's caught by higher level code.